webapp/channels/src/packages/mattermost-redux/src/actions/emojis.test.ts

```diff
@@ -264,36 +264,75 @@ describe('Actions.Emojis', () => {
         expect(state.entities.emojis.nonExistentEmoji.has(missingName)).toBeTruthy();
     });
 
-    it('getCustomEmojisByName', async () => {
-        const testImageData = fs.createReadStream('src/packages/mattermost-redux/test/assets/images/test.png');
-
-        nock(Client4.getBaseRoute()).
-            post('/emoji').
-            reply(201, {id: TestHelper.generateId(), create_at: 1507918415696, update_at: 1507918415696, delete_at: 0, creator_id: TestHelper.basicUser!.id, name: TestHelper.generateId()});
-
-        const {data: created} = await Actions.createCustomEmoji(
-            {
-                name: TestHelper.generateId(),
-                creator_id: TestHelper.basicUser!.id,
-            },
-            testImageData,
-        )(store.dispatch, store.getState) as ActionResult;
-
-        nock(Client4.getBaseRoute()).
-            get(`/emoji/name/${created.name}`).
-            reply(200, created);
-
-        const missingName = TestHelper.generateId();
-
-        nock(Client4.getBaseRoute()).
-            get(`/emoji/name/${missingName}`).
-            reply(404, {message: 'Not found', status_code: 404});
-
-        await Actions.getCustomEmojisByName([created.name, missingName])(store.dispatch, store.getState);
-
-        const state = store.getState();
-        expect(state.entities.emojis.customEmoji[created.id]).toBeTruthy();
-        expect(state.entities.emojis.nonExistentEmoji.has(missingName)).toBeTruthy();
+    describe('getCustomEmojisByName', () => {
+        test('should be able to request a single emoji', async () => {
+            const emoji1 = TestHelper.getCustomEmojiMock({name: 'emoji1', id: 'emojiId1'});
+
+            nock(Client4.getBaseRoute()).
+                post('/emoji/names', ['emoji1']).
+                reply(200, [emoji1]);
+
+            await store.dispatch(Actions.getCustomEmojisByName(['emoji1']));
+
+            const state = store.getState();
+            expect(state.entities.emojis.customEmoji[emoji1.id]).toEqual(emoji1);
+        });
+
+        test('should be able to request multiple emojis', async () => {
+            const emoji1 = TestHelper.getCustomEmojiMock({name: 'emoji1', id: 'emojiId1'});
+            const emoji2 = TestHelper.getCustomEmojiMock({name: 'emoji2', id: 'emojiId2'});
+
+            nock(Client4.getBaseRoute()).
+                post('/emoji/names', ['emoji1', 'emoji2']).
+                reply(200, [emoji1, emoji2]);
+
+            await store.dispatch(Actions.getCustomEmojisByName(['emoji1', 'emoji2']));
+
+            const state = store.getState();
+            expect(state.entities.emojis.customEmoji[emoji1.id]).toEqual(emoji1);
+            expect(state.entities.emojis.customEmoji[emoji2.id]).toEqual(emoji2);
+        });
+
+        test('should correctly track non-existent emojis', async () => {
```

Comment:
I wonder if this is in any way better to test by calling twice the `getCustomEmojisByName`, unless the logic to omit the `nonExistentEmoji` is somewhere else. And if so, I wonder if in this function instead.